File: Sample/Source/Config/HotKey.py
# -*- encoding: utf-8 -*-

from Engine import Keyboard
from Source.Scenes import Map


HotKey = {
    Keyboard.Key.Escape: {
        "Scene": Map,
        "Filter": ["casual"],
        "FunctionWhenPressed": Map.openMenu,
        "FunctionWhenReleased": None,
    },
    Keyboard.Key.D: {
        "Scene": Map,
        "Filter": ["casual"],
        "FunctionWhenPressed": None,
        "FunctionWhenReleased": Map.showEnemyBook,
    },
    Keyboard.Key.G: {
        "Scene": Map,
        "Filter": ["casual"],
        "FunctionWhenPressed": None,
        "FunctionWhenReleased": Map.showFloorTeleporter,
    },
}

File: Sample/Scripts/Source/Config/HotKey.lua
local SceneMap = require("Source.Scenes.SceneMap")

return {
    [sf.Keyboard.Key.Escape] = {
        Scene = SceneMap,
        Filter = { "casual" },
        FunctionWhenPressed = SceneMap.openMenu,
        FunctionWhenReleased = nil
    },
    [sf.Keyboard.Key.D] = {
        Scene = SceneMap,
        Filter = { "casual" },
        FunctionWhenPressed = nil,
        FunctionWhenReleased = SceneMap.showEnemyBook
    },
    [sf.Keyboard.Key.G] = {
        Scene = SceneMap,
        Filter = { "casual" },
        FunctionWhenPressed = nil,
        FunctionWhenReleased = SceneMap.showFloorTeleporter
    }
}
